### PR TITLE
Fused implementation of squared loss

### PR DESCRIPTION
**Pull request by saurabhgarg1996**
Fused implementation of squared loss (roughly twice fast when the size of the is large. Here are the benchmarked numbers: 

Screenshot 2023-11-17 at 3 35 32 PM

```
vector squared relu performance:
           size       Triton       Torch
0        4096.0     6.000000    4.800000
1        8192.0    13.714286    9.600000
2       16384.0    31.999999   21.333333
3       32768.0    63.999998   48.000000
4       65536.0   127.999995   85.333330
5      131072.0   219.428568  153.600004
6      262144.0   384.000001  279.272725
7      524288.0   558.545450  409.600010
8     1048576.0   819.200021  534.260858
9     2097152.0   910.222221  481.882362
10    4194304.0  1003.102074  522.893602
11    8388608.0  1068.521715  546.133325
12   16777216.0  1104.539313  558.545450
13   33554432.0  1120.273520  564.965515
14   67108864.0  1129.931030  568.642060
15  134217728.0  1134.822544  570.705364
```

### CODE REVIEW

File: open_lm/activations.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+def relu_squared_backward_kernel(X, dY, dX, N, BLOCK_SIZE: tl.constexpr=1024):
+    idx = tl.program_id(0) * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    mask = idx < N
+    x = tl.load(X + idx, mask=mask, other=0.0)
+    dy = tl.load(dY + idx, mask=mask, other=0.0)
+    dx = tl.where(x > 0, 2 * x * dy, 0.0)
+    tl.store(dX + idx, dx, mask=mask)
+
+# Custom autograd function
+class SquaredReLUFunction(torch.autograd.Function):
```
Comment (achalddave) [inline open_lm/activations.py:25]: Can you add a comment here or in relu_squared_kernel showing the torch code that this is trying to triton-ize? (presumably, torch.nn.ReLU()(x)**2)

Comment (sagadre) [inline open_lm/activations.py:25]: consider putting the triton implementation in a special folder and then just this wrapper class in activations.py. probably ok for now, but might be nice to have all triton stuff in one place

Comment (sagadre) [inline open_lm/activations.py:25]: consider adding a choice here so that squared relu can be tested end-to-end: https://github.com/mlfoundations/open_lm/blob/ba9299b9c566e7147f094f1cda30193e1676bddd/open_lm/params.py#L46C2-L46C2
